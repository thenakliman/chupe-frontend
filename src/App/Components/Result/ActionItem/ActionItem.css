.action-item-container  {
  width: 65rem;
  margin: auto;
}