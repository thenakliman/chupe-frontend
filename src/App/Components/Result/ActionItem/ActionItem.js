import React from 'react';
import propTypes from 'prop-types';

require('./ActionItem.css');

export class ActionItem extends React.Component {
    componentWillMount() {
      this.props.getActionItems();
    }

    render() {
      return (
      <div id='action-item-container-id' className='action-item-container'>
          <div>
            <table id='all-action-items-ordered-list'>
              <thead>
                <tr className='action-item-table-header'>
                  <th> Description </th>
                  <th> Type </th>
                  <th> Deadline </th>
                </tr>
              </thead>
              <tbody id='all-action-item-table-body-id'
                     className={'action-item-table-body'}>
              {
                this.props.actionItems.map((actionItem) => (
                  <tr key={`${actionItem.id}`}
                      id={`action-item-row-id-${actionItem.id}`}
                      className={'selectable-row'}>
                    <td className='action-item-row'>
                            {actionItem.description}
                    </td>
                    <td> {actionItem.type}</td>
                    <td> {actionItem.deadlineToAct}</td>
                  </tr>
                ))
              }
              </tbody>
            </table>
          </div>
      </div>
      );
    }
}

ActionItem.propTypes = {
  actionItems: propTypes.array.isRequired,
  getActionItems: propTypes.func.isRequired,
};
