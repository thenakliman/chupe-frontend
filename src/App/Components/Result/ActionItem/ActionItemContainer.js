import {connect} from 'react-redux';
import {ActionItem} from './ActionItem';

import {getActionItems} from '../../../Actions/actionItemActions';

export const mapStateToProps = (state) => ({
  actionItems: state.actionItems,
});

export const mapDispatchToProps = (dispatch) => ({
  getActionItems: () => dispatch(getActionItems()),
});

export default connect(mapStateToProps, mapDispatchToProps)(ActionItem);
