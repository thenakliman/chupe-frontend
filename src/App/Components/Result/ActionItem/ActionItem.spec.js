/* eslint-disable */
import React from 'react';
import {ActionItem} from './ActionItem';
/* eslint-disable */
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';


describe('ActionItem', () => {
  it('should have action item container', () => {
    const wrapper = shallow(
         <ActionItem getActionItems={jest.fn()}
                actionItems={[]}
         />);

    expect(wrapper.find('#action-item-container-id').length).toBe(1);
  });

  it('should call get action items on mount', () => {
    const getActionItems = jest.fn();
    const wrapper = shallow(
         <ActionItem getActionItems={getActionItems}
                actionItems={[]}
         />);

    expect(getActionItems).toHaveBeenCalledWith();
  });

  it('should match snapshot', () => {
    const actionItems = [{
        description: 'description retro',
        type: 'RETROSPECTION',
        deadlineToAct: '2019-06-05T00:00:00.000+0000'
    }];

    const wrapper = shallow(
        <ActionItem getActionItems={jest.fn()}
                    actionItems={actionItems}
        />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
