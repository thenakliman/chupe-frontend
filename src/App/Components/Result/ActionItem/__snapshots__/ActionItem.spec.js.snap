// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionItem should match snapshot 1`] = `
<div
  className="action-item-container"
  id="action-item-container-id"
>
  <div>
    <table
      id="all-action-items-ordered-list"
    >
      <thead>
        <tr
          className="action-item-table-header"
        >
          <th>
             Description 
          </th>
          <th>
             Type 
          </th>
          <th>
             Deadline 
          </th>
        </tr>
      </thead>
      <tbody
        className="action-item-table-body"
        id="all-action-item-table-body-id"
      >
        <tr
          className="selectable-row"
          id="action-item-row-id-undefined"
          key="undefined"
        >
          <td
            className="action-item-row"
          >
            description retro
          </td>
          <td>
             
            RETROSPECTION
          </td>
          <td>
             
            2019-06-05T00:00:00.000+0000
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
